src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -300,52 +345,111 @@ private static TextSpan GetSpanWithoutAttributes(SyntaxNode node, SyntaxNode roo
         /// with that and only that node. On the other hand placing cursor anywhere in header should still count as selecting the node it's header of.
         /// </para>
         /// </summary>
-        protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts, bool extractParentsOfHeader)
+        protected virtual IEnumerable<SyntaxNode> DefaultNodesExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts, bool extractParentsOfHeader)
         {
             // REMARKS: 
             // The set of currently attempted extractions is in no way exhaustive and covers only cases
             // that were found to be relevant for refactorings that were moved to `TryGetSelectedNodeAsync`.
             // Feel free to extend it / refine current heuristics. 
 
+            foreach (var extractedNode in ExtractNodesSimple(node, syntaxFacts))
+            {
+                yield return extractedNode;
+            }
+
+            if (extractParentsOfHeader)
+            {
+                foreach (var headerNode in ExtractNodesOfHeader(node, syntaxFacts))
+                {
+                    yield return headerNode;
+                    foreach (var extractedNode in ExtractNodesSimple(headerNode, syntaxFacts))
+                    {
+                        yield return extractedNode;
+                    }
```

Comment:
what's the reasoning behind this nested loop?  thanks!

Reply:
can we comment.  i'm not really getting it.

Reply:
That shouldn't have gotten to be committed. It's part of work for next pr (it'll actually get changed, to something different).

Reply:
Generally, I think it's been shown that we can have just one static "fits all "extractor so I'll do some cleanup on that whole thing with next PR.

Reply:
Also header extraction gets problematic with nodes (when holes are to be concerned (local declaration with initializers)) so I'll it'll probably get changed to work on tokens.